### DISCUSSION TITLE
Add better DNS error management and healthchecks

### DISCUSSION THREAD

**Discussion by fxthomas** [containers/podman#23169] [Ideas]
In rootfull podman using the `netadvark` backend:

```bash
# podman network create test
# podman run --network test -it debian /bin/bash
```

Inside the container, DNS seems to be setup:

```
root@91e9c212fe35:/# cat /etc/resolv.conf 
search dns.podman
nameserver 10.89.0.1
```

...but no query actually succeeds. A transient systemd unit is created which I assume should do DNS resolution, and it's actually listed as "running", but the logs say otherwise:

```
# systemctl status run-r45dbb35d230b4f1bba27920c1ec360b0.scope
● run-r45dbb35d230b4f1bba27920c1ec360b0.scope - /usr/lib/podman/aardvark-dns --config /run/containers/networks/aardvark-dns -p 53 run
     Loaded: loaded (/run/systemd/transient/run-r45dbb35d230b4f1bba27920c1ec360b0.scope; transient)
  Transient: yes
     Active: active (running) since Tue 2024-07-02 22:42:08 CEST; 7min ago
 Invocation: c36a19007bcd465a98420d81b667deec
      Tasks: 1 (limit: 4314)
     Memory: 492K (peak: 1M)
        CPU: 9ms
     CGroup: /system.slice/run-r45dbb35d230b4f1bba27920c1ec360b0.scope
             └─1303791 /usr/lib/podman/aardvark-dns --config /run/containers/networks/aardvark-dns -p 53 run

Jul 02 22:42:08 xxxxx systemd[1]: Started /usr/lib/podman/aardvark-dns --config /run/containers/networks/aardvark-dns -p 53 run.
Jul 02 22:42:08 xxxxx aardvark-dns[1303791]: Unable to start server: unable to start CoreDns server: Address already in use (os error 98)
```

I'm not sure if the fact that aardvark fails is a bug -- it's probably my mistake, the "Address already in use" message is probably because I'm running a DNS server on the host so port 53 is already taken (see https://github.com/containers/aardvark-dns/issues/13 for more discussion).

On the other hand, a nice feature for `podman` should be to report any errors from its subcomponents without having to dig up an obscure systemd unit. What do you think?

**Comment by Luap99** [comment]
Yes the bind call should happen before the fork in aardvark-dns so that the error can be reported properly. That is definitely on the TODO list
